libminifi/include/processors/GetFile.h

```diff
@@ -101,23 +113,17 @@ class GetFile : public core::Processor {
   // Put full path file name into directory listing
   void putListing(std::string fileName);
   // Poll directory listing for files
-  void pollListing(std::queue<std::string> &list, int maxSize);
+  void pollListing(std::queue<std::string> &list,const GetFileRequest &request);
   // Check whether file can be added to the directory listing
-  bool acceptFile(std::string fullName, std::string name);
+  bool acceptFile(std::string fullName, std::string name, const GetFileRequest &request);
+  // Get file request object.
+  GetFileRequest request_;
   // Mutex for protection of the directory listing
+
   std::mutex mutex_;
-  std::string _directory;
-  bool _recursive;
-  bool _keepSourceFile;
-  int64_t _minAge;
-  int64_t _maxAge;
-  int64_t _minSize;
-  int64_t _maxSize;
-  bool _ignoreHiddenFile;
-  int64_t _pollInterval;
-  int64_t _batchSize;
-  uint64_t _lastDirectoryListingTime;
-  std::string _fileFilter;
+
+  std::map<std::string,std::atomic<uint64_t>> last_listing_times_;
```

Comment:
Not sure I follow the usage of the map.  Presumably we only have one directory per instance as there is no input.